### ISSUE TITLE
Group orders

### ISSUE THREAD

**Issue by mark-boute** [KiOui/TOSTI#529]
Group orders based on users and sort queue based on the first order time.
This would remove the case where someone is clicking their Tosti at the same speed my grandpa uses his phone results in a lot of in-between orders. This simulates the old-fashioned method of serving customers customer at a time.

```
1a          1a
2a    ===>  1b
2b          2a
1b          2b
```

**Comment by JobDoesburg** [comment]
The current behavior is intended and @KiOui and I actually discussed it a lot (considering both the intended behavior but also the maintainability and elegance of the software).

I think this could cause some unwanted edge cases if there is a large delay between the orders. What should happen if there's 15 minutes in-between the orders?

I agree that you might want to do this, but only for a 1 minute window or something.

**Comment by mark-boute** [comment]
https://github.com/KiOui/TOSTI/blob/eea789964266e38dd62fbb4e18438b6b2f345f28/website/orders/api/v1/views.py#L50C1-L56C10

@JobDoesburg I think this order_by should be changed, is that assumption correct?

**Comment by JobDoesburg** [comment]
Yes so it seems. You probably want to group on user and sort on the min order time of each group. But only group if the order times are at most x minutes apart, the latter of which I wouldn't directly know how to implement

**Comment by KiOui** [comment]
I would maybe even go as far and just implement this in the javascript on the frontend.

**Comment by JobDoesburg** [comment]
> I would maybe even go as far and just implement this in the javascript on the frontend.

Noooooooo

**Comment by mark-boute** [comment]
> I would maybe even go as far and just implement this in the javascript on the frontend.

I think you want to stay as close as possible to the source of truth here, hence change it right where that truth is generated
